baseline_quest/decompose/analyze_decomposer.py

```diff
@@ -0,0 +1,140 @@
+# coding=utf-8
+# Copyright 2018 The Google AI Language Team Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A script to calculate Recall @ |Pred| for the QUEST dataset, where |Pred| is 
+the arbitrary size of the predicted document list (no truncation).
+It prints the recall and list size for each individual query with the full 
+query text, and saves the output to a human-readable text file in a 'results' 
+subdirectory.
+"""
+
+import json
+from typing import Dict, Iterable, List
+import numpy as np
+import os
+
+# --- Hardcoded File Paths ---
+GOLD_FILE_PATH = "../../../data/train_subset.jsonl"
+PRED_FILE_PATH = "pred_retrieved_set_ops_limited.jsonl" 
+OUTPUT_REPORT_PATH = "recall_report_decomposer_limited.txt" 
+
+
+def read_jsonl(path: str) -> Iterable[Dict]:
+    """Reads a JSONL file line by line."""
+    with open(path, "r", encoding="utf-8") as f:
+        for line in f:
+            line = line.strip()
+            if line:
+                try:
+                    yield json.loads(line)
+                except json.JSONDecodeError:
+                    print(f"Warning: Skipping malformed line in {path}")
+                    continue
+
+def main():
```

Comment:
General nit: Maybe it's better to expose these functionalities as function calls rather than main().